x/launch/keeper/msg_server_settle_request_test.go

```diff
@@ -0,0 +1,268 @@
+package keeper
+
+import (
+	"fmt"
+	"testing"
+
+	codec "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/stretchr/testify/require"
+	spnerrors "github.com/tendermint/spn/pkg/errors"
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/launch/types"
+)
+
+func TestMsgSettleRequest(t *testing.T) {
+	var (
+		invalidChain, _          = sample.ChainID(0)
+		addr1                    = sample.AccAddress()
+		k, pk, srv, _, sdkCtx, _ = setupMsgServer(t)
+		ctx                      = sdk.WrapSDKContext(sdkCtx)
+		chains                   = createNChain(k, sdkCtx, 2)
+		requests                 = createNRequest(k, sdkCtx, 6)
+	)
+	chains[0].LaunchTriggered = true
+	k.SetChain(sdkCtx, chains[0])
+	chains[1].ChainID = "foo"
+	k.SetChain(sdkCtx, chains[1])
+	tests := []struct {
+		name string
+		msg  types.MsgSettleRequest
+		want uint64
+		err  error
+	}{
+		{
```

Comment:
We should have one case where the `applyRequest` fails to ensure the message fails if the request application fails.

Just one case since all the cases are tested in `TestApplyRequest`

Reply:
we already have all failures cases for the `applyRequest` method

and also all cases for msg failures:
https://github.com/tendermint/spn/pull/235/files#diff-ab80113e0f271d47bc52c73c901a3c0bafdb4d2ef481d8f5ca63a2a011e1c152R13

Reply:
Yes all the failure cases for request application should be in `TestApplyRequest`

We should still assert that this specific message fails if `applyRequest` fails

Reply:
done on `b0ddecc`